framework/src/utils/MooseEnumItem.C

```diff
@@ -0,0 +1,68 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+// MOOSE includes
+#include "MooseEnumItem.h"
+#include "MooseUtils.h"
+
+MooseEnumItem::MooseEnumItem(const std::string & name, const int & id)
+  : _raw_name(MooseUtils::trim(name)),
+    _name(MooseUtils::toUpper(_raw_name)),
+    _id(id),
+    _unique(std::hash<std::string>()(std::to_string(_id) + _name))
```

Comment:
I was thinking about this last night and realized that using a hash in an enumeration doesn't make any sense. The very definition of enumeration means to number something. The IDs are guaranteed to be numeric and unique (by the base class implementation and error checks). If I have an enumeration of X=0, Y=1, Z=2 and I ask the enumeration to sort them, they'll be scrambled. This is insane.